app/api/domains/osu.py

```diff
@@ -1667,6 +1667,18 @@ async def register_account(
     # are safe for registration.
     errors: Mapping[str, list[str]] = defaultdict(list)
 
+    # Disable in-game registration if enabled
+    if app.settings.DISABLE_INGAME_REGISTRATION:
+        errors["password"].append(
+            "The in-game registration is disabled. Please register on the website.",
+        )
+        errors = {k: ["\n".join(v)] for k, v in errors.items()}
+        errors_full = {"form_error": {"user": errors}}
```

Comment:
since we're now doing this in 2 places it might be nice to extract this into a common method

Reply:
I think this can be simplified, since it only returns one single error

Reply:
maybe we can have a constant, something like this:

```py
INGAME_REGISTRATION_DISALLOWED_ERROR = {"form_error": {"user": {"password": ["In-game registration is disabled. Please register on the website."]}}}
```

and then just return that as the content

Reply:
Check my commit

Reply:
it would be clearer in intentions imo, you had the same idea as me just you didn't move it into a constant

Reply:
it seems a bit... too weird to have this as a constant to me.

if we're doing that i'd rather store generic responses somewhere and have smth like
```py
INGAME_REGISTRATION_DISALLOWED_RESPONSE = ORJSONResponse(...)
```

Reply:
yeah, maybe. i just think it reads weird in-lined.